tremor-pipeline/src/op/qos/wal.rs

```diff
@@ -248,12 +248,14 @@ impl WAL {
         Ok(events)
     }
 
-    fn store_event(&mut self, uid: u64, mut event: Event) -> Result<()> {
+    fn store_event(&mut self, source_id: u64, mut event: Event) -> Result<()> {
         let id = self.wal.generate_id()?;
         let write: [u8; 8] = unsafe { mem::transmute(id.to_be()) };
-        event.id.add_id(uid, id); // = Ids::new(uid, id);
+        // TODO: figure out if handling of separate streams makes sense here
+        // FIXME: this should go into op_meta, shouldnt it?
```

Comment:
cc @Licenser

Reply:
The WAL acknowledges the event downstream (to the source) so it becomes a 'new' event the moment it enters the WAL That was the reason it got a new ID

Reply:
makes sense, now i know what to do here :)